test/routes.js

```diff
@@ -555,3 +555,78 @@ test('error if there are functions defined in the root object', async (t) => {
     t.is(err.message, 'jit is not possible if there are root functions')
   }
 })
+
+test('GET graphiql endpoint', async (t) => {
+  const app = Fastify()
+  app.register(GQL, {
+    graphiql: true
+  })
+
+  const res = await app.inject({
+    method: 'GET',
+    url: '/graphiql'
+  })
+  t.strictEqual(res.statusCode, 302)
+  t.strictEqual(res.headers.location, '/graphiql.html')
+})
+
+test('GET graphiql endpoint with prefix', async (t) => {
+  const app = Fastify()
+  app.register(GQL, {
+    graphiql: true,
+    prefix: '/test-prefix'
+  })
+
+  const res = await app.inject({
+    method: 'GET',
+    url: '/test-prefix/graphiql'
+  })
+
+  t.strictEqual(res.statusCode, 302)
+  t.strictEqual(res.headers.location, '/test-prefix/graphiql.html')
+})
+
+test('GET graphiql endpoint with prefixed wrapper', async (t) => {
+  const app = Fastify()
+  app.register(async function (app, opts) {
+    app.register(GQL, {
+      graphiql: true
+    })
+  }, { prefix: '/test-wrapper-prefix' })
+
+  const res = await app.inject({
+    method: 'GET',
+    url: '/test-wrapper-prefix/graphiql'
+  })
+
+  t.strictEqual(res.statusCode, 302)
+  t.strictEqual(res.headers.location, '/test-wrapper-prefix/graphiql.html')
+})
+
+test('GET graphql endpoint with prefix', async (t) => {
+  const app = Fastify()
+  app.register(GQL, {
+    prefix: '/test-prefix'
+  })
+
+  const res = await app.inject({
+    method: 'GET',
+    url: '/test-prefix/graphql'
```

Comment:
Can you please add a proper GraphQL query here?

Reply:
Ok I will change that.

But the test shouldn't be focus on only one feature ? Adding a query could make the test fail for another reason than its original purpose.  When I wrote this test, I was wondering if I should make a graphql query here, but I ended up only using a simple GET as a 400 response is enough to ensure that the endpoint was reached.

Reply:
You are right, as it’s better to test one thing only. I consider a 400 an error path, not a success path, and we should really validate the happy path in this situation.

Reply:
Oh well I see your point, it's the 4xx as it's an error this could be returned from a hook or something like that and it doesn't ensure the validation of the desired path. 
Ok, so I will change the test with the add query !